### PR TITLE
update demo notebook tests

### PR DESCRIPTION
**Pull request by cehbrecht**
## Overview

This PR is a follow-up of PR #159.

Changes:

* Activated tests of demo notebooks on travis.
* Moved non-working notebooks to examples.
* Updated pytest nbval output-sanitize ... ordered by notebook.
* Skipped owslib-esgfwps from birdy requirements (only used in notebook examples)

## Related Issue / Discussion

#159

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -59,7 +59,7 @@ test: ## run tests quickly with the default Python
 
 test-nb: ## run tests quickly with the default Python
 	#py.test for notebooks
-	pytest --nbval $(CURDIR)/notebooks/demo --sanitize-with $(CURDIR)/notebooks/output_sanitize.cfg --ignore $(CURDIR)/notebooks/*/.ipynb_checkpoints
```
Comment (Zeitsperre) [inline Makefile]: The reason for keeping this is to ensure that if the Jupyter kernel creates checkpoints. those won't be tested. Just want to be sure that this change is desired.

Comment (cehbrecht) [reply Makefile]: The behavior of `--ignore` was not as expected. When I try it locally it scans *all* notebooks (demo, examples, ...) except those in the ignore list. Without the ignore at least it worked.

Comment (cehbrecht) [reply Makefile]: ... maybe `--ignore` conflicts with the `--nbval` strategy. Looks like if I use `--nbval` only it ignores `.ipynb_checkpoints` by default.

Comment (Zeitsperre) [reply Makefile]: Awesome. I imagine that that's a new feature. Looks good to me.

**Comment by cehbrecht** [comment]
@Zeitsperre thanks for review :)

**Pull request merged**